### DISCUSSION TITLE
A way to invalidate a family provider.

### DISCUSSION THREAD

**Discussion by cc-nogueira** [rrousselGit/riverpod#1498] [Q&A] [answered]
I need to invalidate a family provider (invalidate all providers of a family) when the application loads a new project. Can't use ref.invalidate(...) because Family is not a ProviderBase extension.

Is there another way of doing this (besides re-assigning the provider variable)?

Thanks!

**Comment by rrousselGit** [comment, marked as answer]
There's no way to do that. That would be anti-pattern as you'd quickly end-up refreshing more than you need.

The correct way to do this is to keep track of the currently used `family` arguments. And to do something among the lines of:

```dart
for (final id in ref.read(parametersForMyFamilyProvider)) {
  ref.invalidate(myFamily(id));
}
```

**Reply by scopendo** [reply]
As an aside, I see that `invalidate` was added in release [2.0.0-dev.4](https://pub.dev/packages/riverpod/versions/2.0.0-dev.9/changelog#200-dev4).  I wondered what the rationale was for not making this the behaviour of `refresh`.
